x-pack/plugins/integration_assistant/public/components/create_integration/create_integration_assistant/steps/data_stream_step/sample_logs_input.tsx

```diff
@@ -90,31 +104,36 @@ const parseLogsContent = (
       samplesFormat = { name: 'ndjson', multiline: false };
     }
   } catch (parseNDJSONError) {
+    if (parseNDJSONError instanceof RangeError) {
+      return { error: i18n.LOGS_SAMPLE_ERROR.TOO_LARGE_TO_PARSE };
+    }
     try {
       const { entries, pathToEntries, errorNoArrayFound } = parseJSONArray(fileContent);
       if (errorNoArrayFound) {
-        return { error: i18n.LOGS_SAMPLE_ERROR.NOT_ARRAY, logSamples: [] };
+        return { error: i18n.LOGS_SAMPLE_ERROR.NOT_ARRAY };
       }
       parsedContent = entries;
       samplesFormat = { name: 'json', json_path: pathToEntries };
     } catch (parseJSONError) {
+      if (parseJSONError instanceof RangeError) {
+        return { error: i18n.LOGS_SAMPLE_ERROR.TOO_LARGE_TO_PARSE };
+      }
       try {
         parsedContent = parseNDJSON(fileContent, true);
         samplesFormat = { name: 'ndjson', multiline: true };
       } catch (parseMultilineNDJSONError) {
+        if (parseMultilineNDJSONError instanceof RangeError) {
+          return { error: i18n.LOGS_SAMPLE_ERROR.TOO_LARGE_TO_PARSE };
+        }
         return {
           logSamples: fileContent.split('\n').filter((line) => line.trim() !== ''),
         };
```

Comment:
This is the last case when everything we tried has failed, could we add a comment explaining the reason for this return? It's not obvious that we are leaving the `samplesFormat` _undefined_ to trigger the _analyzeLogs_ task in the backend.

Reply:
Yes, please. I know it's from a previous PR (that I did not review). But since we are adding changes here it would be great to explain this return in the final catch block, it is a bit confusing. It took me a while to understand why it is not returning `error`.

Reply:
Sure, I've added the comment, improved the function docs and updated this PR's description ;)

Reply:
Thank you